keras/backend/tensorflow_backend.py

```diff
@@ -2486,13 +2486,27 @@ def __init__(self, inputs, outputs,
             raise ValueError('Some keys in session_kwargs are not '
                              'supported at this '
                              'time: %s', session_kwargs.keys())
+        self._extra_fetches_info = dict()
         self._callable_fn = None
-        self._merged_summaries = None
         self._feed_arrays = None
         self._feed_symbols = None
         self._symbol_vals = None
         self._session = None
 
+    def _add_fetch(self, fetch, fetch_callback):
+        fetch_index = len(self.fetches)
+        self.fetches.append(fetch)
+        self._extra_fetches_info[fetch_index] = fetch_callback
+        if self._callable_fn is not None:
+            self._callable_fn = None
+        return fetch_index
+
+    def _remove_fetch(self, fetch_index):
+        self.fetches.pop(fetch_index)
+        self._extra_fetches_info.pop(fetch_index)
+        if self._callable_fn is not None:
```

Comment:
You don't need to worry if `None` overrides `None`, so you can safely remove the conditional and only keep the attribution.

Reply:
Ah yes, the check was only needed when making the callable there.